UltraStar Play/Assets/Scenes/PartyMode/PartyModeManager.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UIElements;
+using UniInject;
+using UniRx;
+using Random = UnityEngine.Random;
+
+// Disable warning about fields that are never assigned, their values are injected.
+#pragma warning disable CS0649
+
+// Singleton that handles data during a Party Mode game
+
+public class PartyModeManager
```

Comment:
- By convention, USPlay uses `Manager` suffix only for singleton MonoBehaviour classes.
- I'd prefer calling this `PartyModeControl`